test/node.js

```diff
@@ -37,6 +41,94 @@ test('node pass on nodeOptions', async t => {
 	t.is(stdout, 'foo');
 });
 
+test.serial(
+	'node removes --inspect from nodeOptions when defined by parent process',
+	async t => {
+		const originalArgv = process.execArgv;
+		process.execArgv = ['--inspect', '-e'];
+		const {stdout, stderr} = await execa.node('console.log("foo")', {
+			stdout: 'pipe'
+		});
+		process.execArgv = originalArgv;
+
+		t.is(stdout, 'foo');
+		t.is(stderr, '');
+	}
+);
+
+test.serial(
+	'node removes --inspect=9222 from nodeOptions when defined by parent process',
+	async t => {
+		const originalArgv = process.execArgv;
+		process.execArgv = ['--inspect=9222', '-e'];
+		const {stdout, stderr} = await execa.node('console.log("foo")', {
+			stdout: 'pipe'
+		});
+		process.execArgv = originalArgv;
+
+		t.is(stdout, 'foo');
+		t.is(stderr, '');
+	}
+);
+
+test.serial(
+	'node removes --inspect-brk from nodeOptions when defined by parent process',
+	async t => {
+		const originalArgv = process.execArgv;
+		process.execArgv = ['--inspect-brk', '-e'];
+		const childProc = execa.node('console.log("foo")', {
+			stdout: 'pipe'
+		});
+		process.execArgv = originalArgv;
+
+		setTimeout(() => {
+			childProc.cancel();
+		}, 1000);
+
+		const {stdout, stderr} = await childProc.catch(error => error);
+
+		t.is(stdout, 'foo');
+		t.is(stderr, '');
+	}
+);
+
+test.serial(
+	'node removes --inspect-brk=9222 from nodeOptions when defined by parent process',
+	async t => {
+		const originalArgv = process.execArgv;
+		process.execArgv = ['--inspect-brk=9222', '-e'];
+		const childProc = execa.node('console.log("foo")', {
+			stdout: 'pipe'
+		});
+		process.execArgv = originalArgv;
+
+		setTimeout(() => {
+			childProc.cancel();
+		}, 1000);
+
+		const {stdout, stderr} = await childProc.catch(error => error);
+
+		t.is(stdout, 'foo');
+		t.is(stderr, '');
+	}
+);
+
+test.serial(
+	'node should not remove --inspect when passed through nodeOptions',
+	async t => {
+		const {stdout, stderr} = await execa.node('console.log("foo")', {
+			stdout: 'pipe',
+			nodeOptions: ['--inspect', '-e']
+		});
+
+		t.is(stdout, 'foo');
+		t.regex(
+			stderr,
+			/^Debugger listening on ws:\/\/127.0.0.1:9229\/.*[\r\n]+For help, see: https:\/\/nodejs.org\/en\/docs\/inspector$/
```

Comment:
```suggestion
			/Debugger listening/
```

I don't want to have to change it if it slightly changes on the Node.js side.

Reply:
This would also allow using `string.includes()` directly, which might be simpler:

```js
t.true(stderr.includes('Debugger listening'));
```